.circleci/scripts/changelog-check.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+#
+# This script should run on every PR to post a comment to the PR if there is no
+# entry in .changelog associated with the PR and the PR does not have the `pr/no-changelog` label.
+
+# Requires $CIRCLE_PROJECT_USERNAME, $CIRCLE_PROJECT_REPONAME, and $CIRCLE_PULL_REQUEST from CircleCI
+
+set -o pipefail
+
+# colorized status prompt
+function status {
+    tput setaf 4
+    echo "$@"
+    tput sgr0
+}
+
+# Exit early if we are not running against a PR
+if [ -z "$CIRCLE_PULL_REQUEST" ]; then
+    status "changelog-check: This workflow was not part of a PR"
+    exit 0
+fi
+
+pr_number=$(basename "$CIRCLE_PULL_REQUEST")
+resp=$(curl -f -s -H "Authorization: token ${GITHUB_TOKEN}" "https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${pr_number}")
+ret="$?"
+if [[ "$ret" -ne 0 ]]; then
+    status "The GitHub API returned $ret which means it was probably rate limited."
+    exit $ret
+fi
+
+base_branch=$(echo "$resp" | jq -r '.base.ref')
+labels=$(echo "$resp" | jq --raw-output '.labels[] | .name')
```

Comment:
I think by using github actions we can get this information without an API call by using the github context `github.event.pull_request.base.ref` and `github.event.pull_request.labels`.